crates/rolldown_plugin_import_glob/src/lib.rs

```diff
@@ -82,52 +82,91 @@ pub struct GlobImportVisit<'ast, 'a> {
-                  let mut files = vec![];
-                  // import.meta.glob('./dir/*.js')
-                  // import.meta.glob(['./dir/*.js', './dir2/*.js'])
-
-                  let mut opts = ImportGlobOptions::default();
-                  match call_expr.arguments.as_slice() {
-                    [first] => self.eval_glob_expr(first, &mut files),
-                    // import.meta.glob('./dir/*.js', { import: 'setup' })
-                    [first, second] => {
-                      self.eval_glob_expr(first, &mut files);
-                      extract_import_glob_options(second, &mut opts);
-                    }
-                    [first, second, _rest @ ..] => {
-                      self.eval_glob_expr(first, &mut files);
-                      extract_import_glob_options(second, &mut opts);
-                    }
-                    [] => {}
-                  }
-
-                  // generate:
-                  //
-                  // {
-                  //   './dir/ind.js': __glob__0_0_,
-                  //   './dir/foo.js': () => import('./dir/foo.js'),
-                  //   './dir/bar.js': () => import('./dir/bar.js').then((m) => m.setup),
-                  // }
-                  *expr = self.generate_glob_object_expression(&files, &opts, call_expr.span);
-                  self.current += 1;
-                }
-              }
-              _ => {}
-            }
-          }
-        }
-        _ => {}
+    if let Some(new_expr) = self.maybe_visit_obj_call(expr) {
+      *expr = new_expr;
+    } else if let Some(new_expr) = self.maybe_visit_glob_import_call(expr, false, false) {
+      *expr = new_expr;
+    } else {
+      walk_mut::walk_expression(self, expr);
+    }
+  }
+}
+
+impl<'ast> GlobImportVisit<'ast, '_> {
+  fn maybe_visit_obj_call(&mut self, expr: &Expression<'ast>) -> Option<Expression<'ast>> {
+    let Expression::CallExpression(expr) = expr else { return None };
+    let Expression::StaticMemberExpression(callee) = &expr.callee else { return None };
+    if callee.property.name != "keys" && callee.property.name != "values" {
+      return None;
+    }
+    let Expression::Identifier(i) = &callee.object else { return None };
+    // TODO: check is_global_identifier_reference
+    if i.name != "Object" {
+      return None;
+    }
+    let [arg] = expr.arguments.as_slice() else { return None };
+    let new_expr = self.maybe_visit_glob_import_call(
+      arg.as_expression()?,
+      callee.property.name != "keys",
+      callee.property.name == "keys",
+    )?;
+
+    Some(self.ast_builder.expression_call(
+      expr.span,
+      Expression::from(self.ast_builder.member_expression_static(
+        callee.span,
+        self.ast_builder.expression_identifier(i.span, "Object"),
+        callee.property.clone(),
+        false,
+      )),
+      NONE,
+      self.ast_builder.vec1(Argument::from(new_expr)),
+      false,
+    ))
```

Comment:
Is there a way to only mutate `expr.arguments` instead of recreating the entire `CallExpression`? I can't figure it out.

Reply:
Never mind, I figured it out :)